File: web/src/app/shared/Models/quiz.ts
export class Quiz {
    QuizId: number = 0;
    SubjectId: number = 0;
    Title: string = '';
    Time: number = 0;
    Details: string = '';
    Opened: Date = new Date();
    Due: Date =  new Date();
}

export class QuizModel {
    quizId: number = 0;
    subjectId: number = 0;
    title: string = '';
    details: string = '';
    opened: Date = new Date();
    due: Date =  new Date();
    showScore: boolean;
    time: number = 0;
}

export class QuizModelShowScore {
    quizId: number = 0;
    showScore: boolean;
}

File: E-LearnignWebAPI/ElearningBO/E-Learning/Quiz.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ElearningBO.E_Learning
{
    public class Quiz
    {
        [Key]
        public int QuizId { get; set; }

        [ForeignKey("Subject")]
        public int SubjectId { get; set; }
        public string Title { get; set; }
        public string Details { get; set; }
        public DateTime Opened { get; set; }
        public DateTime Due { get; set; }
        public bool isDelete { get; set; }
        public bool showScore { get; set; }
        public int Time { get; set; }
        public virtual Subject Subject { get; set; }
    }
}
